packages/in_app_purchase/in_app_purchase/README.md

```diff
@@ -184,18 +204,45 @@ Both underlying stores handle consumable and non-consumable products differently
 you're using `InAppPurchase`, you need to make a distinction here and
 call the right purchase method for each type.
 
+<?code-excerpt "example/lib/main.dart (MakePurchase)"?>
 ```dart
-final ProductDetails productDetails = ... // Saved earlier from queryProductDetails().
-final PurchaseParam purchaseParam = PurchaseParam(productDetails: productDetails);
-if (_isConsumable(productDetails)) {
-  InAppPurchase.instance.buyConsumable(purchaseParam: purchaseParam);
+late PurchaseParam purchaseParam;
+
+if (Platform.isAndroid) {
+  // NOTE: If you are making a subscription purchase/upgrade/downgrade, we recommend you to
+  // verify the latest status of you your subscription by using server side receipt validation
+  // and update the UI accordingly. The subscription purchase status shown
+  // inside the app may not be accurate.
+  final GooglePlayPurchaseDetails? oldSubscription =
+      _getOldSubscription(productDetails, purchases);
+
+  purchaseParam = GooglePlayPurchaseParam(
+      productDetails: productDetails,
+      changeSubscriptionParam: (oldSubscription != null)
+          ? ChangeSubscriptionParam(
+              oldPurchaseDetails: oldSubscription,
+              prorationMode:
+                  ProrationMode.immediateWithTimeProration,
+            )
+          : null);
+} else {
+  purchaseParam = PurchaseParam(
+    productDetails: productDetails,
+  );
+}
+
+if (productDetails.id == _kConsumableId) {
```

Comment:
`_isConsumable` made the example easier to understand without context, so we should keep that. It's fine to introduce it into the example app.